cli/util/util.go

```diff
@@ -938,3 +940,28 @@ func CopyFileDeep(src, dst string) error {
 	}
 	return copy.Copy(src, dst)
 }
+
+// TsStrToIprotoTs transforms string with number or RFC339Nano time
+// to IProto timestamp string.
+func TsStrToIprotoTs(input string) (string, error) {
+	// Float.
+	floatTimestamp, err := strconv.ParseFloat(input, 64)
+	if err == nil {
+		return strconv.FormatFloat(floatTimestamp, 'f', -1, 64), nil
+	}
+	// RFC3339/RFC3339Nano = "2006-01-02T15:04:05.999999999Z07:00"
+	rfc3339NanoTs, err := time.Parse(time.RFC3339Nano, string(input))
+	if err == nil {
+		tsSec := rfc3339NanoTs.Unix()
+		tsNanoSec := rfc3339NanoTs.Nanosecond()
+		ts := fmt.Sprintf("%s.%s", strconv.FormatInt(tsSec, 10),
+			strconv.FormatInt(int64(tsNanoSec), 10))
+		return ts, nil
+	}
+	// Trigger an error.
+	if input != "" {
+		return "", err
+	}
+	// Default value.
+	return strconv.FormatUint(math.MaxUint64, 10), nil
```

Comment:
The code structure is confusing. I think it would be better.
```suggestion
	if input == "" {
		// Default value.
		return strconv.FormatUint(math.MaxUint64, 10), nil
	}

	// Float.
	floatTimestamp, err := strconv.ParseFloat(input, 64)
	if err == nil {
		return strconv.FormatFloat(floatTimestamp, 'f', -1, 64), nil
	}

	// RFC3339/RFC3339Nano = "2006-01-02T15:04:05.999999999Z07:00"
	rfc3339NanoTs, err := time.Parse(time.RFC3339Nano, string(input))
	if err != nil {
		return "", err
	}

	tsSec := rfc3339NanoTs.Unix()
	tsNanoSec := rfc3339NanoTs.Nanosecond()
	ts := fmt.Sprintf("%s.%s", strconv.FormatInt(tsSec, 10), strconv.FormatInt(int64(tsNanoSec), 10))
	return ts, nil
```

Reply:
Don't fully agree about `confusing`. But I moved default value handling to the beginning.